src/Microsoft.DotNet.Darc/src/DarcLib/AzureDevOpsClient.cs

```diff
@@ -413,43 +561,54 @@ public async Task<IList<Check>> GetPullRequestChecksAsync(string pullRequestUrl)
                         new Check(
                             checkState,
                             status["context"]["name"].ToString(),
-                            $"{pullRequestUrl}/{status["id"]}"));
+                            $"https://dev.azure.com/{accountName}/{projectName}/{statusesPath}/{status["id"]}"));
                 }
             }
 
             return statuses;
         }
 
-        public async Task<string> GetPullRequestBaseBranch(string pullRequestUrl)
-        {
-            HttpResponseMessage response = await this.ExecuteGitCommand(HttpMethod.Get, pullRequestUrl, _logger);
-
-            JObject content = JObject.Parse(await response.Content.ReadAsStringAsync());
-            string baseBranch = content["sourceRefName"].ToString();
-            const string refsHeads = "refs/heads/";
-            if (baseBranch.StartsWith(refsHeads))
+        /// <summary>
+        ///     Execute a command on the remote repository.
+        /// </summary>
+        /// <param name="method">Http method</param>
+        /// <param name="accountName">Azure DevOps account name</param>
+        /// <param name="projectName">Project name</param>
+        /// <param name="requestPath">Path for request</param>
+        /// <param name="logger">Logger</param>
+        /// <param name="body">Optional body if <paramref name="method"/> is Put or Post</param>
+        /// <param name="versionOverride">API version override</param>
+        /// <returns>Http response</returns>
+        private async Task<HttpResponseMessage> ExecuteRemoteGitCommand(
+            HttpMethod method,
+            string accountName,
+            string projectName,
+            string requestPath,
+            ILogger logger,
+            string body = null,
+            string versionOverride = null)
+        {
+            using (HttpClient client = CreateHttpClient(accountName, projectName, versionOverride))
```

Comment:
>CreateHttpClient [](start = 39, length = 16)

It's a little... odd... to make a new HttpClient to set a bunch of "base" stuff. Why not just create the request with the right values on it instead of all the base things on a single use HttpClient? That would allow sharing the HttpClients if we wanted to later, and also means all the information is in one place (the HttpRequestMessage) instead of split across two (the client and the message), which should make testing/refactoring easier.

Reply:
@ChadNedzlek I think the reason that @jcagme did it this way is that the client configuration looks different between GitHub and Azure DevOps (in headers, base addresses, etc.) but he still wanted to share the request manager code.